src/services/repo-index/repo-index.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from "@nestjs/common";
+import { Connection, Repository } from "typeorm";
+
+import { Logger } from "../../core";
+import { RepoReferenceRecord } from "../../models";
+
+export interface RepoReference {
+  readonly path: string;
+  readonly lastUse: number;
+  readonly lastFetch?: number;
+}
+
+const FETCH_CACHE_EXPIRY = 30_000; // 30s;
+
+@Injectable()
+export class RepoIndexService {
+  private logger = new Logger(RepoIndexService);
+  private readonly repos = new Map<string, RepoReference>();
+  private repository: Repository<RepoReferenceRecord>;
+
+  constructor(connection: Connection) {
+    this.repository = connection.getRepository(RepoReferenceRecord);
+    this.init().catch(e => {
+      this.logger.error("Failed to load data from database", e);
+    });
+  }
+
+  public async init() {
```

Comment:
>init [](start = 15, length = 4)

any reason to have this public ? seems that the constructor calls it

Reply:
this one actually no